hbase-common/pom.xml

```diff
@@ -97,6 +97,14 @@
       <groupId>io.opentelemetry</groupId>
       <artifactId>opentelemetry-semconv</artifactId>
     </dependency>
+    <!--
+      we depend on hadoop-common and hadoop-common depends on curator and
+      zookeeper, so we need to include netty4 dependency here
+    -->
+    <dependency>
```

Comment:
Sorry, is there a reason this can't come in transitively?

I wonder if we should enable maven dependency analyzer plugin, which would disallow dependencies that aren't used directly (and require explicit dependencies for classes we do use)

Reply:
There is a nasty issue for netty that, some dependencies will use netty-all while some others may use something like netty-handler, maven does not know that netty-all includes netty-handler, so it will treat them as different dependencies and then cause runtime class conflicts. So here we will exclude all netty dependencies from our libraries, and then manually include netty-all. You can see what we have done in the root pom file.

Reply:
Oh, the newest netty-all module does not include any real classes, just depend on all other netty modules...
But anyway, it could still lead to different versions of netty modules if we rely on transitive dependencies of netty.

Reply:
I think we should include netty-bom with scope import and the version we want. This will add all netty modules to our dependency management with the same version. So then any transitive dependencies that depend on any netty module will all use the same version.

Reply:
Sounds good. We could try this for netty 4.

Reply:
And maybe netty3 is also OK if we have a netty3 dependency in the dependencyManagement section. Let me try to remove the exclusion.